plugins/Feedback/vue/src/FeedbackQuestion/FeedbackQuestion.vue

```diff
@@ -0,0 +1,173 @@
+    </div>
+    <div class="ratefeature" >
+      <MatomoDialog
+        v-model="showFeedbackForm"
+        @validation = "sendFeedback()"
+      >
+        <div
+          class="ui-confirm ratefeatureDialog"
+        >
+          <h2>{{ translate(`Feedback_Question${question}`) }}</h2>
+          <p
+            v-html="translate('Feedback_FeedbackSubtitle',
+            `<i class='icon-heart red-text'></i>`)"></p>
+          <br/>
+          <div class="messageContainer">
+            <div class="error-text" v-if="errorMessage">{{ errorMessage }}</div>
+            <textarea id="message" :class="{'has-error':errorMessage}" v-model="feedbackMessage"/>
+          </div>
+          <br/>
+          <p
+            v-html="translate('Feedback_Policy',`<a rel='nofollow' href='https://matomo.org/privacy-policy/' target='_blank'>`,'</a>')"></p>
+          <input
+            type="button"
+            role="validation"
+            :value="translate('Feedback_SendFeedback')"
+          />
+          <input
+            type="button"
+            role="cancel"
+            :value="translate('General_Cancel')"
+          />
+        </div>
+      </MatomoDialog>
+      <MatomoDialog v-model="feedbackDone">
+        <div
+          class="ui-confirm ratefeatureDialog"
+        >
+          <h2>{{ translate(`Feedback_ThankYou`) }}</h2>
+          <p v-html="translate('Feedback_ThankYourForFeedback',
+        `<i class='icon-heart red-text'></i>`)">
+          </p>
+          <input
+            type="button"
+            role="cancel"
+            :value="translate('General_Close')"
+          />
+        </div>
+      </MatomoDialog>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+import {
+  MatomoDialog, AjaxHelper, setCookie, getCookie, translate,
+} from 'CoreHome';
+
+const { $ } = window;
+
+const cookieName = 'feedback-question';
+export default defineComponent({
+
+  props: {
+    showQuestionBanner: String,
+  },
+  components: {
+    MatomoDialog,
+  },
+  computed: {
+    isHidden() {
+      if (this.showQuestionBanner === '0') {
+        return true;
+      }
+      return !!this.hide;
+    },
+  },
+  data() {
+    return {
+      questionText: '',
+      question: 0,
+      hide: null,
+      feedbackDone: false,
+      expanded: false,
+      showFeedbackForm: false,
+      feedbackMessage: null,
+      errorMessage: null,
+    };
+  },
+  watch: {
+    showFeedbackForm(val) {
+      // eslint-disable-next-line no-underscore-dangle
+      this.questionText = translate(`Feedback_Question${this.question}`);
+      if (val) {
+        setInterval(() => {
+          $('#message').focus();
+        }, 500);
+      }
+    },
+  },
+  created() {
+    if (this.showQuestionBanner !== '0') {
+      // eslint-disable-next-line no-restricted-globals
+      if (location.href.includes('forceFeedbackTest')) {
+        this.question = 0;
+        setCookie(cookieName, 0, 7);
+      } else {
```

Comment:
I guess this one is to ensure the UI test always shows the same question? Guess this could be achieved also by setting the cookie in the UI test using [`page.setCookie`](https://pptr.dev/#?product=Puppeteer&version=v10.2.0&show=api-pagesetcookiecookies)

Reply:
@sgiehl I tried set page.setCookie in the Test, but its return method was not found. Then I tried to add `setCookies` in `tests/lib/screenshot-testing/support/page-renderer.js` still no luck

Reply:
You may need to add `setCookie` to the list of proxied methods here: https://github.com/matomo-org/matomo/blob/4.x-dev/tests/lib/screenshot-testing/support/page-renderer.js#L53
If that method is not proxied it will only be available through `page.webpage.setCookie`